Testscripts/Windows/Power-Hibernate.ps1

```diff
@@ -292,19 +294,41 @@ install_package "ethtool"
 		if ($vfname -ne '') {
 			$tx_queue_count2 = Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "ethtool -l ${vfname} | grep -i tx | tail -n 1 | cut -d ':' -f 2 | tr -d '[:space:]'" -runAsSudo
 			$interrupt_count2 = Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "cat /proc/interrupts | grep -i mlx | grep -i msi | wc -l" -runAsSudo
-		}
-		if ($tx_queue_count1 -ne $tx_queue_count2) {
-			Write-LogErr "Before hibernation, Tx queue count - $tx_queue_count1. After waking up, Tx queue count - $tx_queue_count2"
-			throw "Tx queue counts changed after waking up."
-		} else {
-			Write-LogInfo "Successfully verified Tx queue count matching in Current hardware settings."
-		}
 
-		if ($interrupt_count1 -ne $interrupt_count2) {
-			Write-LogErr "Before hibernation, MSI interrupts of mlx driver - $interrupt_count1. After waking up, MSI interrupts of mlx driver - $interrupt_count2"
-			throw "MSI interrupt counts changed after waking up."
+			if ($tx_queue_count1 -ne $tx_queue_count2) {
+				Write-LogErr "Before hibernation, Tx queue count - $tx_queue_count1. After waking up, Tx queue count - $tx_queue_count2"
+				throw "Tx queue counts changed after waking up."
+			} else {
+				Write-LogInfo "Successfully verified Tx queue count matching in Current hardware settings."
+			}
+
+			if ($interrupt_count1 -ne $interrupt_count2) {
+				Write-LogErr "Before hibernation, MSI interrupts of mlx driver - $interrupt_count1. After waking up, MSI interrupts of mlx driver - $interrupt_count2"
+				throw "MSI interrupt counts changed after waking up."
+			} else {
+				Write-LogInfo "Successfully verified MSI interrupt counts matching"
+			}
+
+			# Verify the TX/RX packets keep increasing after waking up
+			$tx_packets_first = Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "ethtool -S ${vfname} | grep tx_packets: | awk '{print `$2}'" -runAsSudo
+			$rx_packets_first = Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "ethtool -S ${vfname} | grep rx_packets: | awk '{print `$2}'" -runAsSudo
+			Start-Sleep -s 3
+			$tx_packets_second = Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "ethtool -S ${vfname} | grep tx_packets: | awk '{print `$2}'" -runAsSudo
+			$rx_packets_second = Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "ethtool -S ${vfname} | grep rx_packets: | awk '{print `$2}'" -runAsSudo
+			if ($tx_packets_first -ge $tx_packets_second) {
+				Write-LogErr "First collected TX packets: $tx_packets_first. Second collected TX packets: $tx_packets_second"
```

Comment:
Traffic is not running, so it does not make sense.

Reply:
Hi @juhlee-microsoft ,
The VF NIC is the slave of the primary NIC so that there's always traffic even if I don't send traffic additionally. I checked it manually and the number is indeed increasing.

Reply:
I know those polling packets of VF from PF, but 3-second is too short. It means this test might be risky. Can we increase 10 to 30 seconds for sure?

Reply:
Oh got it. Thanks! I changed it to 10 seconds. If we found it's still too short we can make it longer then. Is it acceptable?